src/lib/reducers/apps.js

```diff
@@ -58,19 +58,37 @@ export const fetchApps = () => async dispatch => {
   }
 }
 
-const setHomeApp = appsList => async dispatch => {
+/**
+ *
+ * @param {Array} appsList
+ */
+const setDefaultApp = appsList => async dispatch => {
   return stack.get
     .context()
     .then(context => {
-      const homeLink =
+      const defaultRedirection =
         context.data &&
         context.data.attributes &&
         context.data.attributes.default_redirection
-      const slugRegexp = /^([^/]+)\/.*/
-      const homeSlug =
-        homeLink && homeLink.match(slugRegexp) && homeLink.match(slugRegexp)[1]
-      if (!homeSlug) return appsList
-      const homeApp = appsList.find(app => app.slug === homeSlug)
+      let homeApp = null
+      //self hosted cozy has no context by default
+      //so let's use hardcoded home slug if needed
+      if (!defaultRedirection) {
+        const HOME_APP_SLUG = 'home'
+        homeApp = appsList.find(app => app.slug === HOME_APP_SLUG)
+      } else {
+        const slugRegexp = /^([^/]+)\/.*/
+        const defaultAppSlug =
+          defaultRedirection &&
+          defaultRedirection.match(slugRegexp) &&
+          defaultRedirection.match(slugRegexp)[1]
+        homeApp = appsList.find(app => app.slug === defaultAppSlug)
+      }
+
+      if (!homeApp) {
+        return appsList
+      }
```

Comment:
Nit: I am a bit uncertain here what's the effect of not having the final dispatch is. It seems a bit awkward to return an array here where we return the result of the dispatch otherwise. Are we sure we need to return something ? The function is called `set` so AMHA it should only have a side effect and not return anything.